starknet_py/tests/e2e/conftest.py

```diff
@@ -434,7 +436,53 @@ def compiled_proxy(request) -> str:
     """
     Returns source code of compiled proxy contract
     """
-    return (contracts_compiled_dir / request.param).read_text("utf-8")
+    return read_compiled_contract(request.param)
+
+
+@pytest.fixture(name="custom_proxy")
+def custom_proxy() -> str:
+    """
+    Returns compiled source code of a custom proxy
+    """
+    return read_compiled_contract("oz_proxy_custom_compiled.json")
+
+
+@pytest.fixture(name="old_proxy")
+def old_proxy() -> str:
+    """
+    Returns compiled (using starknet-compile 0.8.1) source code of OpenZeppelin's proxy using address and delegate_call.
+    """
+    return read_compiled_contract("_oz_proxy_address_0.8.1_compiled.json")
+
+
+@pytest_asyncio.fixture(name="deploy_proxy_to_contract")
+async def deploy_proxy_to_contract(request, gateway_account_client) -> DeployResult:
+    """
+    Declares a contract and deploys a proxy pointing to that contract.
+    """
+    compiled_proxy_name, compiled_contract_name = request.param
+    compiled_proxy = read_compiled_contract(compiled_proxy_name)
+    compiled_contract = read_compiled_contract(compiled_contract_name)
+
+    declare_tx = make_declare_tx(compiled_contract=compiled_contract)
+    declare_result = await gateway_account_client.declare(declare_tx)
+
+    implementation_key = (
+        "implementation_hash"
+        if "implementation_hash" in compiled_proxy
+        else "implementation"
+    )
+    deployment_result = await Contract.deploy(
+        compiled_contract=compiled_proxy,
+        constructor_args={
+            implementation_key: declare_result.class_hash,
+            "selector": get_selector_from_name("put"),
+            "calldata": [69, 420],
+        },
+        client=gateway_account_client,
+    )
+    await deployment_result.wait_for_acceptance()
+    return deployment_result
```

Comment:
test module now contains a `utils.py` file. Let's make this a normal function and move it there since this fixture isn't really used as a fixture.